src/mmw/js/src/draw/views.js

```diff
@@ -131,11 +130,141 @@ function validatePointWithinDataSourceBounds(latlng, dataSource) {
     return d.promise();
 }
 
-var DrawWindow = Marionette.ItemView.extend({
+var DrawWindow = Marionette.LayoutView.extend({
     template: windowTmpl,
 
     id: 'draw-window',
 
+    regions: {
+        selectTypeRegion: '#select-area-region',
+        drawRegion: '#draw-region',
+        watershedDelineationRegion: '#place-marker-region'
+    },
+
+    ui: {
+        'selectBoundary': '#select-boundary-button',
+        'drawArea': '#draw-area-button',
+        'delineateWatershed': '#delineate-watershed-button',
+        'uploadFile': '#file-upload-button',
+        'resetDraw': '.reset-draw-button'
+    },
+
+    events: {
+        'click @ui.selectBoundary': 'toggleSelectBoundary',
+        'click @ui.drawArea': 'toggleDrawArea',
+        'click @ui.delineateWatershed': 'toggleDelineateWatershed',
+        'click @ui.uploadFile': 'toggleUploadFile',
+        'click @ui.resetDraw': 'resetDrawingState'
+    },
+
+    modelEvents: {
+        'change:selectedDrawTool': 'onShow'
+    },
+
+    constants: {
+        selectedDrawTool: 'selectedDrawTool',
+        selectBoundary: 'selectBoundary',
+        drawArea: 'drawArea',
+        delineateWatershed: 'delineateWatershed',
+        uploadFile: 'uploadFile'
+    },
+
+    templateHelpers: function() {
+        return {
+            activeDrawTool: this.model.get(this.constants.selectedDrawTool),
+        };
+    },
+
+    initialize: function() {
+        var map = App.getLeafletMap(),
+            ofg = L.featureGroup();
+        this.model.set('outlineFeatureGroup', ofg);
+        this.model.set(this.constants.selectedDrawTool, null);
+        map.addLayer(ofg);
+        this.rwdTaskModel = new models.RwdTaskModel();
+    },
+
+    onShow: function() {
+        var drawTools = settings.get('draw_tools'),
+            activeDrawTool = this.model.get(this.constants.selectedDrawTool);
+
+        // Update the view to mount the proper region divs
+        this.render();
```

Comment:
I deleted a comment on an earlier commit, but something still seems fishy about calling `render` in `onShow`. I think if you call `render` on the `modelChange` event it will call before `onShow`, and you won't need to call `render` again here.